win32ss/gdi/ntgdi/freetype.c

```diff
@@ -206,6 +378,65 @@ BYTE FASTCALL IntCharSetFromCodePage(UINT uCodePage)
     return DEFAULT_CHARSET;
 }
 
+static BOOL
+SubstituteFontRecurse(LOGFONTW* pLogFont);
+
+static __inline VOID
+FindBestFontFromList(FONTOBJ **FontObj, ULONG *MatchPenalty,
+                     const LOGFONTW *LogFont,
+                     const PLIST_ENTRY Head);
+
+static BOOL
+MatchFontName(PSHARED_FACE SharedFace, PUNICODE_STRING Name1, FT_UShort NameID, FT_UShort LangID);
+
+static BOOL
+FontLink_GetReady(
+    _Inout_ PFONTLINK pFontLink)
+{
+    FONTOBJ *pFontObj;
+    ULONG MatchPenalty;
+    UNICODE_STRING Name1;
+    PPROCESSINFO Win32Process;
+
+    if (pFontLink->bIgnoreThis)
+        return FALSE;
+
+    if (pFontLink->SharedFace)
+        return TRUE;
+
+    MatchPenalty = 0xFFFFFFFF;
+    pFontObj = NULL;
+
+    // Search private fonts
+    Win32Process = PsGetCurrentProcessWin32Process();
+    ASSERT_FREETYPE_LOCK_HELD();
```

Comment:
It the `FontLink_GetReady` has to be called with the freetype lock held, you could move this assert at the top of the function.

Reply:
Moved.